code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -106,3 +106,196 @@
+		final_block_chance = 0
+	return ..()
+
+/datum/enchantment
+	/// used for blackbox logging
+	var/name = "You shouldn't be seeing this, file an issue report."
+	/// used for wizards/cultists examining the runes on the blade
+	var/desc = "Someone messed up, file an issue report."
+	/// used for damage values
+	var/power = 0
+	/// whether the enchant procs despite not being in proximity
+	var/ranged = FALSE
+	/// stores the world.time until it can be used again, the `initial(cooldown)` is the cooldown between activations.
+	var/cooldown = -1
+
+/datum/enchantment/proc/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	if(world.time < cooldown)
+		return FALSE
+	if(!istype(target))
+		return FALSE
+	if(target.stat == DEAD)
+		return FALSE
+	if(!ranged && !proximity)
+		return FALSE
+	cooldown = world.time + initial(cooldown)
+	return TRUE
+
+/datum/enchantment/proc/on_gain(obj/item/melee/spellblade, mob/living/user)
+
+/datum/enchantment/lightning
+	name = "lightning"
+	desc = "this blade conducts arcane energy to arc between its victims"
+	power = 20
+	cooldown = 3 SECONDS
+
+/datum/enchantment/lightning/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	. = ..()
+	if(.)
+		zap(target, user, list(user), power)
+
+
+/datum/enchantment/lightning/proc/zap(mob/living/target, mob/living/source, protected_mobs, voltage)
+	source.Beam(target, "lightning[rand(1,12)]", 'icons/effects/effects.dmi', time = 2 SECONDS, maxdistance = 7, beam_type = /obj/effect/ebeam/chain)
+	if(!target.electrocute_act(voltage, flags = SHOCK_TESLA)) // if it fails to shock someone, break the chain
+		return
+	protected_mobs += target
+	addtimer(CALLBACK(src, .proc/arc, target, voltage, protected_mobs), 2.5 SECONDS)
+
+/datum/enchantment/lightning/proc/arc(mob/living/source, voltage, protected_mobs)
+	voltage = voltage - 4
+	if(voltage <= 0)
+		return
+
+	for(var/mob/living/L in oview(7, get_turf(source)))
+		if(L in protected_mobs)
+			continue
+		zap(L, source, protected_mobs, voltage)
+		break
+
+/datum/enchantment/fire
+	name = "fire"
+	desc = "this blade will self immolate on hit, releasing a ball of fire. it also makes the weilder immune to fire"
+	power = 20
+	cooldown = 8 SECONDS
+
+/datum/enchantment/fire/on_gain(obj/item/melee/spellblade/S, mob/living/user)
+	..()
+	RegisterSignal(S, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED), .proc/toggle_traits)
+	if(user)
+		toggle_traits(S, user)
+
+/datum/enchantment/fire/proc/toggle_traits(obj/item/I, mob/living/user)
+	if(HAS_TRAIT_FROM(user, TRAIT_NOFIRE, MAGIC_TRAIT))
+		REMOVE_TRAIT(user, TRAIT_NOFIRE, MAGIC_TRAIT)
+		REMOVE_TRAIT(user, TRAIT_RESISTHEAT, MAGIC_TRAIT)
+	else
+		ADD_TRAIT(user, TRAIT_RESISTHEAT, MAGIC_TRAIT)
+		ADD_TRAIT(user, TRAIT_NOFIRE, MAGIC_TRAIT)
+
+/datum/enchantment/fire/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	. = ..()
+	if(.)
+		fireflash_s(target, 4, 400 * power, 500)
```

Comment:
the `power`s of the enchantment being all over the place is super weird. Especially if you still feel the need to multiply that raw `power` by a constant when passing to lower level procs.
Either standardize the base power of all of them to be the exact same value (and remove the per-specialization overrides), and multiply them by the appropriate amount when passing to `fireflash_s` and similar, OR remove this sort of multiplication OR add a readable comment on each of the specialization that translates the number to english description of what it means ("[power] fire damage", "[power] seconds weakened on hit" etc)